server/container/guice/mailbox/pom.xml

```diff
@@ -46,6 +46,10 @@
             <groupId>${james.groupId}</groupId>
             <artifactId>apache-james-mailbox-event-memory</artifactId>
         </dependency>
+        <dependency>
+            <groupId>${james.groupId}</groupId>
+            <artifactId>apache-james-mailbox-store</artifactId>
```

Comment:
Well I didn't have a choice here as it was using the dependency that mailbox-adapter had to mailbox-store to use some mailbox-store classes. It's in `server-container-guice-mailbox` though so I think we need it? But I'm not 100% sure, so please give me a feedback on this one

Reply:
As you say `it was using the dependency that mailbox-adapter` so it's not a blocker to the adoption to me.

Now comes question time: What do we need `mailbox-store` for in here ? (exhaustive list please)

Reply:
Interesting work in #3342 , might be worth trying a combine? 

So in `james-server-guice-mailbox` it is what you fixed in #3342 

However you have modules implementing this one that are using stuff from `mailbox-store`:

* `james-server-jpa-guice`
* `james-server-memory-guice`
* `james-server-cassandra-guice`

But that would allow at least at first not to expose mailbox-store to the other modules being used by `james-server-guice-mailbox`

And I think I can do a little better here too, it might be only the 3 above modules only using `mailbox-adapter` as well, not necessary to be declared here

Reply:
```
However you have modules implementing this one that are using stuff from mailbox-store:

    james-server-jpa-guice
    james-server-memory-guice
    james-server-cassandra-guice

```

It is normal these modules are specific to mailbox implementation that relies on `mailbox-store`.

> But that would allow at least at first not to expose mailbox-store to the other modules being used by james-server-guice-mailbox

This module is not bound to a given mailbox implementation, removing the mailbox-store dependency is a good move.

> Interesting work in #3342 , might be worth trying a combine?

If you like leaving dangerously, then yes for sure :-)

> And I think I can do a little better here too, it might be only the 3 above modules only using mailbox-adapter as well, not necessary to be declared here

:+1: :-)